mlrun/api/crud/artifacts.py

```diff
@@ -33,40 +33,78 @@ def store_artifact(
         self,
         db_session: sqlalchemy.orm.Session,
         key: str,
-        data: dict,
-        uid: str,
+        artifact: dict,
+        object_uid: str = None,
         tag: str = "latest",
         iter: int = 0,
-        project: str = mlrun.mlconf.default_project,
+        project: str = None,
+        producer_id: str = None,
     ):
         project = project or mlrun.mlconf.default_project
         # In case project is an empty string the setdefault won't catch it
-        if not data.setdefault("project", project):
-            data["project"] = project
+        if not artifact.setdefault("project", project):
+            artifact["project"] = project
 
-        if data["project"] != project:
+        if artifact["project"] != project:
             raise mlrun.errors.MLRunInvalidArgumentError(
-                f"Artifact with conflicting project name - {data['project']} while request project : {project}."
-                f"key={key}, uid={uid}, data={data}"
+                f"Artifact with conflicting project name - {artifact['project']} while request project : {project}."
```

Comment:
```suggestion
                f"Conflicting project name - storing artifact with project {artifact['project']} into a different project : {project}."
```
Clearer error message.